src/analysis/processing/qgsalgorithmshortestline.cpp

```diff
@@ -0,0 +1,180 @@
+/***************************************************************************
+                         qgsalgorithmshortestline.cpp
+                         ---------------------
+    begin                : September 2021
+    copyright            : (C) 2020 by Matteo Ghetta, Clemens Raffler
+    email                : clemens dot raffler at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+//Disclaimer:This feature was originally developed in Python by: Matteo Ghetta, August 2021
+
+#include "qgsalgorithmshortestline.h"
+#include "qgsdistancearea.h"
+
+///@cond PRIVATE
+
+QString QgsShortestLineAlgorithm::name() const
+{
+  return QStringLiteral( "shortestline" );
+}
+
+QString QgsShortestLineAlgorithm::displayName() const
+{
+  return QObject::tr( "Shortest line between layers" );
+}
+
+QStringList QgsShortestLineAlgorithm::tags() const
+{
+  return QObject::tr( "distance,shortest,minimum,nearest" ).split( ',' );
+}
+
+QString QgsShortestLineAlgorithm::group() const
+{
+  return QObject::tr( "Vector analysis" );
+}
+
+QString QgsShortestLineAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectoranalysis" );
+}
+
+QString QgsShortestLineAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates a line layer as the "
+                      "shortest line between the source and the destination layer. "
+                      "By default only the first nearest feature of the "
+                      "destination layer is taken into account. "
+                      "The n-nearest neighboring features number can be specified. "
+                      "If a maximum distance is specified, then only "
+                      "features which are closer than this distance will "
+                      "be considered. The output features will contain all the "
+                      "source layer attributes, all the attributes from the n-nearest "
+                      "feature and the additional field of the distance. "
+                      "This algorithm uses purely Cartesian calculations for distance, "
+                      "and does not consider geodetic or ellipsoid properties when "
+                      "determining feature proximity."
```

Comment:
Best note here that despite this the distance attribute IS calculated using ellipsoidal distances.